lib/fluent/plugin/filter_process_snmptrap.rb

```diff
@@ -59,68 +19,22 @@ def start
     end
 
     def filter(tag, time, record)
-      @time = time
-      @tag = tag
-      message = record.clone
-      record["machineId"] = ""
-      record["event"] = ""
-      record["status"] = ""
-      record["device"] = ""
-      record["severity"] = ""
-      record["sensorValue"] = ""
-      record["error"] = ""
-      record["message"] = ""
-
-      determineMachineId(record)
-      processEvent(record)
-      determineSensorValue(record)
-      determineStatus(record)
-      record["message"] = message
-      record.delete_if { |key, value| key.to_s.match(/(?:SNMPv2-(\w+)(::)(\w+)((\.)(\d+)){1,13}|(host))/)}
-      return record
-    end
-
-    def determineMachineId(record)
-      rmcSerialNo = record[@@rmcSerialNum].to_s
-      if rmcSerialNo.nil?
-        record["error"] << " : Can not determine Machine ID"
-        return
-      end
-      record["machineId"] = "HPE:#{coloregion}:#{rmcSerialNo}"
-    end
 
-    def processEvent(record)
-      record["severity"] = "info"
-      determineEvent(record)
-      if record["event"].nil?
-        record["error"] << " : Can not determine the event"
-      end
-    end
+      # Replace invalid characters
+      fixedRecord = Hash.new
+      record.each { |recKey, recValue|
+          newKey = recKey
+          invalidChars.each { |invalidKey, subChar|
```

Comment:
why is invalidChars in the test file?

Reply:
Because I wanted it to be defined on the config.